### PR TITLE
Add inheritance to FilterCommand

### PR DESCRIPTION
**Pull request by nikkiDEEE**

### CODE REVIEW

**Comment by nikkiDEEE** [comment]
This is just the FilterByTopicCommand

Nic will PR the FilterByImportanceCommand, there wont be any merge conflicts

**Comment by nikkiDEEE** [comment]
Not complete, still refactoring the AddCommand so user can input the importance level

But the comparator works and it can be sorted by importance (when editing the importance in the files itself)

**Comment by nikkiDEEE** [comment]
This PR does not include exception handling for strings cus its giving errors when i attempt to try-catch in the isValidImportance() method in the Parser class... will re-look at it after EG submission 

but for now the importance filter command works and user can input importance from 1-10 and it will give an error prompt if user input anything out of this range in the 'i/' field

File: src/main/java/seedu/clialgo/command/AddCommand.java [hunk 1]
```diff
@@ -28,6 +28,25 @@ public AddCommand(String name, String topic) {
         this.name = name;
         this.topic = topic;
         this.ui = new Ui();
+        importance = 5;
```
Comment (heejet) [inline src/main/java/seedu/clialgo/command/AddCommand.java:31]: Done use magic number. Use a constant instead.

File: src/main/java/seedu/clialgo/command/FilterByImportanceCommand.java [hunk 2]
```diff
@@ -1,7 +1,73 @@
...
     }
+
+    @Override
+    public void printAllTopics(TopicManager topicManager, Ui ui) {
+        ArrayList<CS2040CFile> cs2040cFiles = topicManager.getAllFilesAsFiles();
+        ui.printFilterSuccess();
+        cs2040cFiles.sort(importanceLevel);
+        int serialNumber = 1;
+        for (CS2040CFile cs2040CFile : cs2040cFiles) {
+            System.out.println(serialNumber + ". " + cs2040CFile.getName());
```
Comment (heejet) [inline src/main/java/seedu/clialgo/command/FilterByImportanceCommand.java:33]: I think it is better to print the importance after the name.
E.g
```
1. bubble sort [1]
```

File: src/main/java/seedu/clialgo/command/FilterByImportanceCommand.java [hunk 3]
```diff
@@ -1,7 +1,73 @@
...
+
+    @Override
+    public void printSingleTopic(TopicManager topicManager, Ui ui) {
+        ArrayList<CS2040CFile> cs2040cFiles = topicManager.getOneTopic(this.topic).getCS2040CFilesAsArray();
+        ui.printFilterSuccess();
+        cs2040cFiles.sort(importanceLevel);
+        int serialNumber = 1;
+        System.out.println("[" + this.topic + "]");
+        for (CS2040CFile cs2040cFile : cs2040cFiles) {
+            System.out.println(serialNumber + ". " + cs2040cFile.getName());
```
Comment (heejet) [inline src/main/java/seedu/clialgo/command/FilterByImportanceCommand.java:46]: same as above

File: src/main/java/seedu/clialgo/command/InvalidImportanceCommand.java [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+
+    @Override
+    public void execute(TopicManager topicManager, Ui ui, FileManager fileManager, Buffer buffer) {
+        ui.printInvalidImportance(importance);
+    }
+
+    @Override
+    public boolean equals(Command otherCommand) {
+        return false;
+    }
```
Comment (heejet) [inline src/main/java/seedu/clialgo/command/InvalidImportanceCommand.java:23]: Add in the comparison feature for this

File: src/main/java/seedu/clialgo/file/CS2040CFile.java [hunk 5]
```diff
@@ -20,11 +21,18 @@ public CS2040CFile(String name, String path, String topic, int importance) {
...
+     * @param name Name of the file.
+     * @param path Path to the file.
+     * @param topic Topic that file should be tagged.
+     */
     public CS2040CFile(String name, String path, String topic) {
         this.name = name;
         this.path = path;
         this.topic = topic;
-        this.importance = 5;    // placeholder
+        this.importance = 5;
```
Comment (heejet) [inline src/main/java/seedu/clialgo/file/CS2040CFile.java:35]: Use constants instead of magic numbers

**Review by heejet — CHANGES_REQUESTED** [review]

**Review by heejet — APPROVED** [review]

**Pull request merged**